### PR TITLE
Nitunit works with groups and markdown files

### PR DESCRIPTION
**Pull request by privat**
A lot of work but quite straightforward. Nit can be a really nice language when doing maintenance of old code.

The first part ot the PR updates the nitdoc to test documentation of groups (as requested by #1201)
The second part of the PR (I planned to do 2 PR but the second was more easy to do than expected) makes that nitunit can also process sand-alone markdown files (documentation, wiki pages, etc.).

Example:

~~~
-- foo/
   |-- README.md
   `-- foo.nit
~~~

~~~sh
# to test all entities of a module (classes, methods, etc.)
$ nitunit foo/foo.nit
# to test all entities of a group (the group and all its modules)
$ nitunit foo
# to test a given markdown file
$ nitunit foo/README.md
~~~

Close: #1201

### CODE REVIEW

**Comment by xymus** [comment]
Nice PR! :shipit:

Can we refer to other nitunits from the same readme file? As I did in the serialization readme?

**Comment by privat** [comment]
@xymus not yet, the extraction of the pieces of code are done without a lot of deep-analysis.
In order to be able to write something like you did, nitunit must be able to understand `module user_credentials` and use it to name the produced temporary file (and thinking about file-name issues, collisions and whatever).

File: share/man/nitunit.md [hunk 1]
```diff
@@ -13,13 +13,16 @@ nitunit [*options*] FILE...
 Unit testing in Nit can be achieved in two ways:
```
Comment (Morriar) [inline share/man/nitunit.md]: So it's three ways now?

File: share/man/nitunit.md [hunk 2]
```diff
@@ -13,13 +13,16 @@ nitunit [*options*] FILE...
...
 
 * using `DocUnits` in code comments
```
Comment (Morriar) [inline share/man/nitunit.md]: Maybe leave "two ways" and change this as

    * using `DocUnits` in code comments or markdown files.

File: share/man/nitunit.md [hunk 3]
```diff
@@ -106,6 +109,14 @@ The `nitunit` command is used to test Nit files:
 
     $ nitunit foo.nit
 
+Groups (repertories) can be given to tests the documentation of the group and of all its Nit files:
```
Comment (Morriar) [inline share/man/nitunit.md]: repertories? like folders?

Comment (Morriar) [inline share/man/nitunit.md]: "to test" *

File: share/man/nitunit.md [hunk 4]
```diff
@@ -106,6 +109,14 @@ The `nitunit` command is used to test Nit files:
...
+
+    $ nitunit lib/foo
+
+Finally, standard markdown documents can be used as is:
```
Comment (Morriar) [inline share/man/nitunit.md]: can be checked*

File: src/loader.nit [hunk 5]
```diff
@@ -448,6 +455,26 @@ redef class ModelBuilder
 		return nmodule
 	end
 
+	# Remove from nit source-files from a list of arguments.
```
Comment (Morriar) [inline src/loader.nit]: I do not understand that synopsys.

File: src/loader.nit [hunk 6]
```diff
@@ -448,6 +455,26 @@ redef class ModelBuilder
...
+	#
+	# Items of `args` that can be loaded as a nit file will be removed from `args` and returned.
```
Comment (Morriar) [inline src/loader.nit]: This line can be a perfect synopsys instead.

File: src/location.nit [hunk 7]
```diff
@@ -55,6 +55,8 @@ class Location
 	var file: nullable SourceFile
 
 	# The starting line number (starting from 1)
+	#
+	# If `line_start==0` then the whole file is considered
```
Comment (Morriar) [inline src/location.nit]: `line_start == 0`, with the spaces at the right place, the sentence is more elegant and easier to read.

File: src/nitunit.nit [hunk 8]
```diff
@@ -65,11 +67,27 @@ var page = new HTMLTag("testsuites")
...
+	if g == null then continue
+	page.add modelbuilder.test_group(g)
+end
+
 for m in mmodules do
 	page.add modelbuilder.test_markdown(m)
 	page.add modelbuilder.test_unit(m)
 end
 
+
```
Comment (Morriar) [inline src/nitunit.nit]: Why a second blank line?

File: src/testing/testing_doc.nit [hunk 9]
```diff
@@ -272,25 +248,65 @@ class NitUnitExecutor
...
 		toolcontext.check_errors
 
 		testsuite.add(tc)
 	end
+
+	# Create and fill the header of a test file `file`.
+	#
+	# The handled on the file is returned and must be completed and closed.
+	#
+	# `file` sould be a valid filepath for a Nit source file.
```
Comment (Morriar) [inline src/testing/testing_doc.nit]: Can I pass a Markdown file?

File: src/testing/testing_doc.nit [hunk 10]
```diff
@@ -385,4 +401,67 @@ redef class ModelBuilder
 
 		return ts
 	end
+
+	# Extracts and executes all the docunits in the readme of the `mgroup`
```
Comment (Morriar) [inline src/testing/testing_doc.nit]: Other comments in this file seem to be on the imperative form.

**Comment by Morriar** [comment]
Quelques comments mais +1 pour le reste.

**Comment by privat** [comment]
small reroll: include comments. thx reviewers.

**Pull request merged**